### DISCUSSION TITLE
After click button very slow get data response?

### DISCUSSION THREAD

**Discussion by zainawan99** [livewire/livewire#4349] [1. Help]
After click button very slow get data response?
![image](https://user-images.githubusercontent.com/24542584/144806623-c72f8eec-21a5-4ae9-8dfa-2e9456297ec2.png)
![image](https://user-images.githubusercontent.com/24542584/144806677-86ecdbcc-0b97-4435-8a52-7ae3c6d51398.png)
![image](https://user-images.githubusercontent.com/24542584/144806820-542823de-3617-47be-9c8c-f9ca5ade014b.png)

**Comment by joshhanley** [comment]
@zainawan99 can you click on the request and take a screenshot of the requests "timing" tab?

It should look like this:

![image](https://user-images.githubusercontent.com/882837/144829258-5a2648ad-fea9-474d-834f-8abf42b6e10d.png)

And also take a screenshot of Laravel Debugbar's queries tab

Looks like this (but with queries)

![image](https://user-images.githubusercontent.com/882837/144829575-72edae69-5380-414f-8bf1-9156d4e73aa5.png)

Thanks!

**Comment by zainawan99** [comment]
yes my queries duplicated 35 times

**Reply by zainawan99** [reply]
![image](https://user-images.githubusercontent.com/24542584/144966046-09582a03-cc67-4fc7-8f4a-12bef563433d.png)

**Reply by joshhanley** [reply]
@zainawan99 I think that is your issue, you need to optimise your app and reduce all the duplicated queries and see how it runs once you've done that 🙂

**Comment by zainawan99** [comment]
but how?

**Comment by joshhanley** [comment]
@zainawan99 you have to go through your queries and ensure they are only being run once.

If you find they are being called multiple times, you need to change your code so it only runs once.

And also make use of eager loading where you can https://laravel.com/docs/8.x/eloquent-relationships#eager-loading
 
Hope this helps!
